skiko/src/awtMain/cpp/windows/OpenGLLibrary.cc

```diff
@@ -0,0 +1,85 @@
+#include <windows.h>
+#include <gl/GL.h>
+#include <jawt_md.h>
+
+typedef void (*PROC_glFinish) (void);
+typedef void (*PROC_glGetIntegerv) (GLenum pname, GLint *data);
+typedef const GLubyte *(*PROC_glGetString) (GLenum name);
+typedef HGLRC (WINAPI * PROC_wglCreateContext) (HDC hDc);
+typedef BOOL (WINAPI * PROC_wglDeleteContext) (HGLRC oldContext);
+typedef PROC (WINAPI * PROC_wglGetProcAddress) (LPCSTR lpszProc);
+typedef BOOL (WINAPI * PROC_wglMakeCurrent) (HDC hDc, HGLRC newContext);
+typedef HGLRC (WINAPI * PROC_wglGetCurrentContext) (void);
+
+class OpenGL32Library {
+public:
+    HINSTANCE lib;
+
+    PROC_glFinish glFinish;
+    PROC_glGetIntegerv glGetIntegerv;
+    PROC_glGetString glGetString;
+    PROC_wglCreateContext wglCreateContext;
+    PROC_wglDeleteContext wglDeleteContext;
+    PROC_wglGetProcAddress wglGetProcAddress;
+    PROC_wglMakeCurrent wglMakeCurrent;
+    PROC_wglGetCurrentContext wglGetCurrentContext;
+
+    OpenGL32Library(HINSTANCE lib) {
+        this->lib = lib;
+        glFinish = (PROC_glFinish) GetProcAddress(lib, "glFinish");
+        glGetIntegerv = (PROC_glGetIntegerv) GetProcAddress(lib, "glGetIntegerv");
+        glGetString = (PROC_glGetString) GetProcAddress(lib, "glGetString");
+        wglCreateContext = (PROC_wglCreateContext) GetProcAddress(lib, "wglCreateContext");
+        wglDeleteContext = (PROC_wglDeleteContext) GetProcAddress(lib, "wglDeleteContext");
+        wglGetProcAddress = (PROC_wglGetProcAddress) GetProcAddress(lib, "wglGetProcAddress");
+        wglMakeCurrent = (PROC_wglMakeCurrent) GetProcAddress(lib, "wglMakeCurrent");
+        wglGetCurrentContext = (PROC_wglGetCurrentContext) GetProcAddress(lib, "wglGetCurrentContext");
+    }
+};
+
+OpenGL32Library* openGL32Library = nullptr;
+
+extern "C"
+{
+    void glFinish(void) {
```

Comment:
Do we really need to examine linker to resolve functions with names that already exists? Can we just call our owns (with prefix or so)?